Forge.TreeWalker/src/TreeWalkerParameters.cs

```diff
@@ -180,8 +192,77 @@ public class TreeWalkerParameters
+        /// Note: Recommended to cache the deserialized ForgeTree in your app and use that to avoid Forge deserializing the JsonSchema each initialize.
+        /// </summary>
+        /// <param name="sessionId">The unique identifier for this session.</param>
+        /// <param name="forgeTree">The ForgeTree for this session.</param>
+        /// <param name="forgeState">The Forge state.</param>
+        /// <param name="callbacks">The callbacks object.</param>
+        /// <param name="token">The cancellation token.</param>
+        public TreeWalkerParameters(
+            Guid sessionId,
+            ForgeTree forgeTree,
+            IForgeDictionary forgeState,
+            ITreeWalkerCallbacksV2 callbacksV2,
+            CancellationToken token)
+        {
+            if (sessionId == Guid.Empty) throw new ArgumentNullException("sessionId");
+            if (forgeTree == null) throw new ArgumentNullException("forgeTree");
+            if (forgeState == null) throw new ArgumentNullException("forgeState");
+            if (callbacksV2 == null) throw new ArgumentNullException("callbacksV2");
+            if (token == null) throw new ArgumentNullException("token");
+
+            this.SessionId = sessionId;
+            this.ForgeTree = forgeTree;
+            this.ForgeState = forgeState;
+            this.CallbacksV2 = callbacksV2;
+            this.Token = token;
+        }
+
+        /// <summary>
+        /// Instantiates a TreeWalkerParameters object with the properties that are required to instantiate a TreeWalkerSession object.
+        /// Note: Recommended to cache the deserialized ForgeTree in your app and use that to avoid Forge deserializing the JsonSchema each initialize.
+        /// </summary>
+        /// <param name="sessionId">The unique identifier for this session.</param>
+        /// <param name="jsonSchema">The JSON schema.</param>
+        /// <param name="forgeState">The Forge state.</param>
+        /// <param name="callbacksV2">The callbacks object.</param>
+        /// <param name="token">The cancellation token.</param>
+        public TreeWalkerParameters(
+            Guid sessionId,
+            string jsonSchema,
+            IForgeDictionary forgeState,
+            ITreeWalkerCallbacksV2 callbacksV2,
+            CancellationToken token)
+        {
+            if (sessionId == Guid.Empty) throw new ArgumentNullException("sessionId");
+            if (string.IsNullOrWhiteSpace(jsonSchema)) throw new ArgumentNullException("jsonSchema");
+            if (forgeState == null) throw new ArgumentNullException("forgeState");
+            if (callbacksV2 == null) throw new ArgumentNullException("callbacksV2");
+            if (token == null) throw new ArgumentNullException("token");
+
+            this.SessionId = sessionId;
+            this.JsonSchema = jsonSchema;
+            this.ForgeState = forgeState;
+            this.CallbacksV2 = callbacksV2;
             this.Token = token;
         }
```

Comment:
Let's remove this constructor. I want folks to use ForgeTree instead of jsonSchema. Introducing a new V2 interface gives a good opportunity to phase jsonSchema out as well.

Reply:
i see. removed.

---
In reply to: [505911437](https://github.com/microsoft/Forge/pull/32#discussion_r505911437) [](ancestors = 505911437)